tests/sles4sap/publiccloud/qesap_terraform.pm

```diff
@@ -0,0 +1,180 @@
+    Default scenario is plain 1 node deployment without HANA and HA stack.
+    Enable 'QESAP_HANA_HA_ENABLED' and adjust 'NODE_COUNT' accordingly for HA/HANA scenario.
+
+=cut
+
+sub set_default_values {
+    set_var_output("QESAP_HANA_HA_ENABLED", "false");
+    set_var_output("NODE_COUNT", 1);
+    set_var_output("QESAP_VM_SIZE", "Standard_E4s_v3");
+    set_var_output("HANA_OS_MAJOR_VERSION", (split("-", get_var("VERSION")))[0]);
+
+    # Cluster related setup
+    my $ha_enabled = get_var("QESAP_HANA_HA_ENABLED") =~ /False|false|0/ ? 0 : 1;
+
+    # set fencing to native if no ha cluster is set to avoid deploying ISCSI server
+    set_var("FENCING_MECHANISM", "native") if $ha_enabled == 1;
+    if ($ha_enabled eq 1 && get_var("NODE_COUNT") > 1) {
+        # set fencing to native if no ha cluster is set to avoid deploying ISCSI server
+        record_info("HA scenario", "Parameters are set for HA/HANA scenario.");
+        set_var_output("USE_SAPCONF", "true");
+        set_var_output("HANA_MEDIA", "/hana/shared/install");
+        set_var_output("INSTANCE_SID", "HA1");
+        set_var_output("INSTANCE_ID", "00");
+        set_var_output("SAP_DOMAIN", "qesap.example.com");
+        set_var("SAP_SIDADM", lc(get_var("INSTANCE_SID") . "adm"));
+    }
+
+    # Cluster needs at least 2 nodes
+    die("HA cluster needs at least 2 nodes. Check 'NODE_COUNT' parameter.") if $ha_enabled eq 1 && get_var("NODE_COUNT") <= 1;
+}
+
+=head2 create_ansible_playbook_list
+
+    Detects HANA/HA scenario from openQA variables and creates "ansible: create:" section in config.yaml file.
+
+=cut
+
+sub create_playbook_section {
+    # Cluster related setup
+    my $ha_enabled = get_var("QESAP_HANA_HA_ENABLED") =~ /False|false|0/ ? 0 : 1;
+    my @playbook_list;
+    my @hana_playbook_list = (
+        "pre-cluster.yaml",
+        "sap-hana-preconfigure.yaml -e use_sapconf=" . get_var("USE_SAPCONF"),
+        "cluster_sbd_prep.yaml",
+        "sap-hana-storage.yaml",
+        "sap-hana-download-media.yaml",
+        "sap-hana-install.yaml",
+        "sap-hana-system-replication.yaml",
+        "sap-hana-system-replication-hooks.yaml",
+        "sap-hana-cluster.yaml"
+    );
+    my $registration = "registration.yaml -e reg_code=" . get_var("SCC_REGCODE_SLES4SAP") . " -e email_address=''";
+
+    # Add registration module - "QESAP_SCC_NO_REGISTER" skips scc registration via ansible
+    push(@playbook_list, $registration) unless get_var("QESAP_SCC_NO_REGISTER");
+    if ($ha_enabled == 1) {
+        push(@playbook_list, @hana_playbook_list);
+    }
+    return (\@playbook_list);
+}
+
+=head2 create_ansible_playbook_list
+
+    Detects HANA/HA scenario from openQA variables and creates "ansible: create:" section in config.yaml file.
+
+=cut
+
+sub create_hana_vars_section {
+    # Cluster related setup
+    my $ha_enabled = get_var("QESAP_HANA_HA_ENABLED") =~ /False|false|0/ ? 0 : 1;
```

Comment:
Same suggestion regarding `i` modifier to regexps.

Reply:
Fixed